src/SmartTransactionsController.ts

```diff
@@ -602,13 +674,13 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
             : originalTxMeta;
 
         if (this.#doesTransactionNeedConfirmation(txHash)) {
-          this.confirmExternalTransaction(
-            txMeta,
+          this.#confirmExternalTransaction(
+            txMeta as TransactionMeta,
             transactionReceipt,
-            baseFeePerGas,
+            baseFeePerGas as Hex,
```

Comment:
This is less than ideal but does seem to align with the current runtime behavior of this code. Maybe we could add TODO comments so that these as assertions are fixed sooner rather than later.

```diff
@@ -675,8 +675,10 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
 
         if (this.#doesTransactionNeedConfirmation(txHash)) {
           this.#confirmExternalTransaction(
+            // TODO: Replace 'as' assertion with correct typing for `txMeta`
             txMeta as TransactionMeta,
             transactionReceipt,
+            // TODO: Replace 'as' assertion with correct typing for `baseFeePerGas`
             baseFeePerGas as Hex,
           );
         }
```